src/database.ts

```diff
@@ -72,42 +76,93 @@ export class Database<D = unknown> {
   /**
    * Execute a set of Statements in batch mode.
    * Batching sends multiple SQL statements inside a single call to the
-   * network. This can have a huge performance impact, as it reduces the
-   * total number of transactions sent to the Tableland smart contract,
-   * thereby reducing gas costs.
+   * network. This can have a huge performance impact, as it only sends
+   * one transaction to the Tableland smart contract, thereby reducing
+   * gas costs.
    * Batched statements are similar to SQL transactions. If a statement
    * in the sequence fails, then an error is returned for that specific
    * statement, and it aborts or rolls back the entire sequence.
    * @param statements A set of Statement objects to batch and submit.
    * @param opts Additional options to control execution.
    * @returns An array of run results.
    */
+  //    Note: if we want this package to mirror the D1 package in a way that
+  //    enables compatability with packages built to exend D1, then the return type
+  //    here will potentially affect if/how those packages work.
+  //    D1-ORM is a good example: https://github.com/Interactions-as-a-Service/d1-orm/
   async batch<T = D>(
     statements: Statement[],
     opts: Signal = {}
-  ): Promise<Array<Result<T>>> {
+    // reads returns an Array with legnth equal to the number of batched statements,
+    // everything else a single result wrapped in an Array for backward compatability.
+    // TODO: In order to work around the Validator API not returning all of the tableIds
+    //       and continuing to work in a backward compatable way, it seems that we have to
+    //       make this type `any` :(
+    //       We should attempt to fix this when the Validator API update happens, or on the
+    //       next major version.
+  ): Promise<any> {
     try {
+      const start = performance.now();
+      // If the statement types are "create" and the statement contains more than one
+      // query (separated by semi-colon) then the sqlparser with throw an Error.
       const normalized = await Promise.all(
         statements.map(async (stmt) => await normalize(stmt.toString()))
       );
+
       const type: string | null = normalized
         .map((stmt) => stmt.type)
         .reduce((a, b): any => (a === b ? a : null));
       if (type == null) {
         throw new Error(
-          "statement error: batch must contain uniform types (e.g., CREATE, INSERT, SELECT, etc)"
+          "statement error: batch must contain uniform types (i.e. one of: create, write, read, acl)"
```

Comment:
Not sure if this makes sense, but since the parser has a specific set of types I thought maybe it made sense to use that nomenclature.
I'm open to keeping the error message the same if anyone has a preference.